tests/Spec/Nri/Ui/Highlighter.elm

```diff
@@ -0,0 +1,315 @@
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> mouseDown "Pothos"
+                |> mouseOver "indirect"
+                |> mouseUp "Pothos"
+                |> ensureMarked [ "Pothos", "", "indirect" ]
+                |> done
+    , test "Highlights element on Space" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> space
+                |> ensureMarked [ "Pothos" ]
+                |> done
+    , test "Removes highlight from element on MouseUp" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> space
+                |> ensureMarked [ "Pothos" ]
+                |> ensureFocusOn "Pothos"
+                |> mouseDown "Pothos"
+                |> mouseUp "Pothos"
+                |> expectViewHasNot
+                    [ Selector.tag "mark" ]
+    , test "Removes entire highlight from a group of elements on MouseUp" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> shiftRight
+                |> releaseShiftRight
+                |> ensureMarked [ "Pothos", "", "indirect" ]
+                |> mouseDown "indirect"
+                |> mouseUp "indirect"
+                |> expectViewHasNot
+                    [ Selector.tag "mark" ]
+    , test "Removes highlight from element on Space" <|
+        \() ->
+            Highlightable.initFragments Nothing "Pothos indirect light"
+                |> program marker
+                |> ensureFocusOn "Pothos"
+                |> space
+                |> ensureMarked [ "Pothos" ]
+                |> ensureFocusOn "Pothos"
+                |> space
+                |> expectViewHasNot
+                    [ Selector.tag "mark" ]
+    ]
+
+
+ensureFocusOn : String -> TestContext marker -> TestContext marker
+ensureFocusOn word textContext =
+    textContext
+        |> ensureView
+            (Query.find [ Selector.attribute (Key.tabbable True) ]
+                >> Query.has [ Selector.text word ]
+            )
+        |> ensureView
+            (Query.findAll [ Selector.attribute (Key.tabbable True) ]
+                >> Query.count (Expect.equal 1)
+            )
+        -- We only manage focus for interactive elements.
+        -- Static elements should not have explicit tabindex -1 as it is redundant,
+        -- they are never included in the tab sequence
+        |> ensureView
+            (Query.findAll [ Selector.attribute (Key.tabbable False) ]
+                >> Query.count (Expect.equal 2)
```

Comment:
I think the assertion about the number of interactive elements belongs in a specific test, rather than a helper. I'd like for `ensureFocusOn` to potentially be usable when there are more than 3 interactive highlightables, you know?